packages/keyboard-shortcuts/src/hooks/use-shortcut.js

```diff
@@ -34,9 +41,9 @@ export default function useShortcut( name, callback, { isDisabled } = {} ) {
 			}
 		}
 
-		shortcuts.current.add( _callback );
+		shortcuts.add( _callback );
 		return () => {
-			shortcuts.current.delete( _callback );
+			shortcuts.delete( _callback );
 		};
-	}, [ name, isDisabled ] );
+	}, [ name, isDisabled, shortcuts ] );
```

Comment:
It good that `shortcuts` ended up being a dependency. If the context value changes, then all shortcuts will be removed from the old registry and added to the new one, automatically.